### ISSUE TITLE
Make configuring learning rate schedulers more flexible

### ISSUE THREAD

**Issue by epwalsh** [allenai/allennlp#4856]
In particular, it would be great if you do something like this:

```jsonnet
"learing_rate_scheduler": [
  // use this scheduler for the first two epochs
  {"type": "linear_with_warmup", "num_epochs": 2},
  // use this scheduler for the remaining epochs
  {"type": "cosine"},
]
```

**Comment by dirkgr** [comment]
I think we'd use one LR scheduler that calls into others to make this as flexible as possible.
